seisgo/noise.py

```diff
@@ -39,238 +40,229 @@ def compute_fft(trace,win_len,step,stainv=None,
+                        tr2.data=np.append(tr2.data,zero)
+                
+                else:
+                    zero=np.zeros(int(abs(egap)//dt))
+                    if egap<0:
+                        tr1.data=np.append(tr1.data,zero)
+                    else:
+                        tr2.data=np.append(tr2.data,zero)
+                
+                tr1_len=tr1.data.shape
+                tr2_len=tr2.data.shape
+                # if traces have the same length, do orientation correction
+                if  tr1_len==tr2_len:
+                    trN,trE=utils.correct_orientations(tr1,tr2,orient)
+                    print('Orientation correction for {} is finished'.format(ista))
+                else:
+                    print('STOP! Error in pading traces. Check start and end times')
+                    print(tr1_start)
+                    print(tr2_start)
+                    print(tr1_end)
+                    print(tr2_end)
+                    print(tr1.data.shape)
+                    print(tr2.data.shape)
+                    print(sfile,ista)
+                    trE,trN=utils.correct_orientations(tr1,tr2,orient)
+                    sys.exit()
+                    
+                #stE=Stream([trE])
+                #stN=Stream([trN])
+                rawdata=[trN,trE]
+
+            # Skip station if it has less than 2 horizontal channels
+            elif rotate_flag==1:
+                print('CONTINUE! Too less chennels for rotation, which requires 2 channels')
+                return None
+            # Skip channel rotation (for vertical channel) if no_rotate_flag==1
+            if no_rotate_flag==1:
+                for tag in channels:
+                    #print(tag,'was passed directly to fft')
+                    tr,ori,sta_name=trace_info(ds,loc,ista,itag,inv1)
+                    rawdata.append(tr)
+        
+        # Skip orientation correction if correct_orientation is False
+        else:
+            for itag in all_tags:
+                if v:print("Assembling raw data for station %s and trace %s" % (ista,itag))
+                # read waveform data
+                tr,ori,sta_name=trace_info(ds,loc,ista,itag,inv1)
+                #tr = ds.waveforms[ista][itag][0]
+                if len(tr.data)!=0:
+                    rawdata.append(tr)
+                else:
+                    print(sfile,'station {0} channel {1} is empty. Continue to the next channel'.format(ista,itag))
+                    continue
+    raw['data']=rawdata
+    raw['inv']=[inv1]
+    raw['coor']=inv1.get_coordinates(sta_name)
+####
+    return raw
     
-                        #exclude some channels in the exclude_chan list.
-                        if comp in exclude_chan:
-                            print(comp+" is in the exclude_chan list. Skip it!")
-                            continue
+
+#assemble FFT with given asdf file name
+def assemble_fft(raw_data,win_len,step,freqmin=None,freqmax=None,
```

Comment:
please make the usage of assemble_fft compatible with the older version. For example, make raw_data could be a list of h5 files. add a line at the very beginning to check. if raw_data is a list of files instead of a dictionary, then call assemble_raw().

Reply:
It is tricky to call assemble_raw() inside assemble_fft() because the input variables are very different.
To accommodate those who have the older version, I can print an instruction if input raw_data is not a dictionary.

Reply:
you can use a simple if statement to check and provide the right operation. 
e.g.,

```python
if isinstance(raw_data,'list'):
      raw_data_out=assemble_raw(raw_data, ...)
      raw_data = raw_data_out
```

Reply:
It is doable, but the problems are:
1. adding additional 9 inputs to assemble_fft()
2. increase execution time
Since it's only used under do_correlation(), I prefer to add a comment to give users instructions

Reply:
I don't understand why you need 9 additional inputs. The goal is: if user run git pull to update their seisgo, they can just run their old scripts without any changes needed. Please try to keep the older way of calling assemble_fft() working. 

I also don't understand why this if statement would increase the execution time. It is just trying to be compatible with the old way of running it in case people have scripts calling it outside of do_correlation().